File: data/multilang_comparison/program_go/src/func35.go
package main
import "math/rand"
var _ = rand.Intn
func Func35(vars *ArrayParam, loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var array30 *Array
   if pCounter > 0 {
       array30 = vars.Data[pCounter-1]
       pCounter--
       array30.RefC++
   } else {
       array30 = &Array{
           Size: 756,
           RefC: 1,
           Id: 30,
           Data: make([]uint32, 756),
       }
   }
   var params0 ArrayParam
   params0.Size = 1
   params0.Data = make([]*Array, params0.Size)
   params0.Data[0] = array30
   var array31 *Array = Func42(&params0, getPath(), loopsFactor);
   params0.Data = nil
   params0.Size = 0
   var loop10 int = 0;
   var loopLimit10 int = (50)/4 + 1;
   for ; loop10 < loopLimit10; loop10++ {
      for i := 0; i < array30.Size; i++ {
          array30.Data[i]++
      }
   }
   for i := 0; i < array31.Size; i++ {
       if array31.Data[i] == 65 {
           return array31
       }
   }
   array30.RefC--
   if array30.RefC == 0 {
       array30.Data = nil
       array30 = nil
   }
   return array31;
}

File: data/multilang_comparison/program_cpp_array/src/func35.cpp
#include "program_cpp_array.hpp" 
Array* func35(Array_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   Array* array30;
   if (pCounter > 0) {
      array30 = vars->data[--pCounter];
      array30->refC++;
      DEBUG_COPY(array30->id);
   } else {
      array30 = new Array();
      array30->size = 756;
      array30->refC = 1;
      array30->id = 30;
      array30->data = new unsigned int[array30->size]();
      DEBUG_NEW(array30->id);
   }
   Array_param params0;
   params0.size = 1;
   Array* data_params0[1];
   params0.data = data_params0;
   params0.data[0] = array30;
   Array* array31 = func42(&params0, get_path(), loopsFactor);
   DEBUG_RETURN(array31->id);
   if(params0.size > 0) {
   	 params0.data.clear();
   }
   unsigned int loop10 = 0;
   unsigned int loopLimit10 = (50)/4 + 1;
   for(; loop10 < loopLimit10; loop10++) {
      for (int i = 0; i < array30->size; i++) {
         array30->data[i]++;
      }
   }
   for (int i = 0; i < array31->size; i++) {
       if (array31->data[i] == 65) {
           return array31;
       }
   }
   array30->refC--;
   if(array30->refC == 0) {
      delete[] array30->data;
      array30->data = nullptr;
      DEBUG_FREE(array30->id);
      delete array30;
   }
   return array31;
}

